packages/server/modules/gendo/services/index.ts

```diff
@@ -11,49 +14,97 @@ import {
   upsertBlobFactory
 } from '@/modules/blobstorage/repositories'
 import { db } from '@/db/knex'
+import { CreateRenderRequest, StoreRender } from '@/modules/gendo/domain/operations'
+import { UploadFileStream } from '@/modules/blobstorage/domain/operations'
+import {
+  getGendoAIAPIEndpoint,
+  getGendoAIKey,
+  getServerOrigin
+} from '@/modules/shared/helpers/envHelper'
+import { GendoRenderRequestError } from '@/modules/gendo/errors/main'
 
 const uploadFileStream = uploadFileStreamFactory({
   upsertBlob: upsertBlobFactory({ db }),
   updateBlob: updateBlobFactory({ db })
 })
 
-export async function createGendoAIRenderRequest(
-  input: GendoAiRenderInput & {
-    userId: string
-    status: string
-    id: string
-    gendoGenerationId?: string
-  }
-) {
-  const baseImageBuffer = Buffer.from(
-    input.baseImage.replace(/^data:image\/\w+;base64,/, ''),
-    'base64'
-  )
-
-  const blobId = crs({ length: 10 })
-  await uploadFileStream(
-    storeFileStream,
-    { streamId: input.projectId, userId: input.userId },
-    {
-      blobId,
-      fileName: `gendo_base_image_${blobId}.png`,
-      fileType: 'png',
-      fileStream: baseImageBuffer
+export const createRenderRequestFactory =
```

Comment:
Refactoring more than in previous PRs (e.g. renamed fn), but none of this shit works anyway